### DISCUSSION TITLE
x-for runs before an x-if that is wrapping it

### DISCUSSION THREAD

**Discussion by AdamSCMM** [alpinejs/alpine#1730] [7. Bugs (archived)]
I have code where an x-if wraps an x-for to stop the x-for from running if the variables are not correct. In v2 this code worked correctly, in v3 the x-for throws an error: `can't access property "length", items is undefined` even though the x-if should be stopping it beforehand.

I've tried a few different ways of writing the if statement, but nothing seems to make a difference

The code looks like:

```
<template x-if="currentDay in days">
or
<template x-if="days[currentDay]">
    <template x-for="(_numbers, _index) in days[currentDay]" :key="_index">
```

You can see a working example at the codepen below. Click 5 or 7 and then null or clear. It breaks when it should not. 5 to 7 works fine

https://codepen.io/acxz/pen/mdmeJRo?editors=1010

It was already suggested to me to add a `|| []` at the end of the for to account for the collection being undefined, but that shouldn't be necessary. This situation should not be occurring like this

**Comment by Pronian** [comment]
I'm not quite sure if I understand the issue but I want to point out that `x-if` and `x-for` templates should have only one child including text nodes.

**Reply by AdamSCMM** [reply]
The code I made the pen from follows that rule, I had added words into the pen to help identify what was where and forgot to account for that issue

The pen has been updated to follow it properly, the issue persists

The issue to me seems to be that the x-for is trying to run even though the wrapping IF statement is false. The For is incorrectly trying to use the set of "items" `days[currentDay]` which is undefined. It should not be trying to do this because IF on `undefined` is false

**Reply by Tim-Wils** [reply]
Looks like this is a bug
